theia-tree-editor/src/browser/tree-editor-widget.tsx

```diff
@@ -132,7 +133,7 @@ export abstract class BaseTreeEditorWidget extends BaseWidget implements Saveabl
         node,
         type,
         property
-    }: AddCommandProperty): void;
+    }: AddCommandProperty): MaybePromise<void>;
```

Comment:
I'd just make this a normal `Promise`. It makes sense that this is handled asynchronously anyway.